apps/server/src/AppController.ts

```diff
@@ -17,8 +17,8 @@ export class AppController {
     return this.appService.getBalance(address);
   }
 
-  @Get('getAllEventsFromBlockNumber')
-  async getAllEventsFromBlockNumber(): Promise<Event[]> {
-    return this.appService.getAllEventsFromBlockNumber();
+  @Get('checkTransactionConfirmationStatus')
+  async checkTransactionConfirmationStatus(@Query('transactionHash') transactionHash: string): Promise<boolean> {
+    return this.appService.checkTransactionConfirmationStatus(transactionHash);
```

Comment:
```suggestion
  @Post('handleTransaction')
  async handleTransaction(@Body?('transactionHash') transactionHash: string): Promise<boolean> {
    return this.appService.handleTransaction(transactionHash);
```
Should be a `POST`, since this can also trigger a 'write' of data.

Down the line we need to properly separate concerns instead of lumping everything in `AppService`